src/editor/components/modals/file-modal/manage-file-sources/file-source-list.jsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect } from "react";
+import styled from "react-emotion";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import format from "date-fns/format";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import DeleteModal from "../../../../../server/components/delete-modal";
+import {
+  deleteFileSource as deleteFileSourceAction,
+  createFileUpdateOperation as createFileUpdateOperationAction,
+  getFileSources as getFileSourcesAction
+} from "../../../../actions/file-source-actions";
+
+import {
+  TextButton,
+  OutlineButton
+} from "../../../../../shared/components/buttons";
+
+import {
+  List,
+  ListItem,
+  ListMain,
+  ListPrimaryText,
+  ListSecondaryText,
+  ListSecondaryTextLink,
+  ListMetadata
+} from "../../../../../server/components/list";
+
+const formatDate = str => format(new Date(str), "MMM dd, uuuu HH:mm:ss");
+
+const FileSourceListContainer = styled.div`
+  width: 100%;
+  margin: auto;
+`;
+
+const FileSourceInterval = styled(ListMetadata)`
+  font-size: 12px;
+  color: gray;
+  padding-left: 6px;
+  padding-right: 6px;
+  margin-left: 4px;
+  margin-right: 4px;
+`;
+
+const NoFileSourcesNotice = styled.span`
+  align-items: center;
+  color: #666;
+  display: grid;
+  font-size: 1.5em;
+  font-weight: bold;
+  text-align: center;
+`;
+
+const FileSourceURLContainer = styled(ListSecondaryText)`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 500px;
+  white-space: nowrap;
+`;
+
+const ClickRunNowToFetch = styled.div`
+  color: #999;
+  font-size: 10px;
+  font-style: italic;
+  align-self: end;
+`;
+
+const FileName = styled.div`
+  font-weight: bold;
+  color: ${({ hasBeenRun }) => (hasBeenRun ? "black" : "#888")};
+`;
+
+const FileSourceURL = styled(ListSecondaryTextLink)`
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+const FileInformationContainer = styled.div`
+  max-width: 500px;
+  display: grid;
+  grid-template-columns: max-content max-content max-content;
+  grid-column-gap: 10px;
+`;
+
+const ListItemCall = styled(ListMetadata)`
+  min-width: 100px;
+  text-align: center;
+  position: relative;
+  height: 100%;
+  display: grid;
+  margin: auto;
+  align-content: center;
+  overflow: hidden;
+  cursor: pointer;
+`;
+
+const Fader = styled.div`
```

Comment:
i think it would be worth breaking something like fader into a reusable component in our component library, something like "SpinnerButton" or whatever you want to call it. it's a nice little thing for giving a bit of feedback, but it seems like a generic thing that doesn't belong in this file.

but instead of inventing something from scratch, perhaps we should build a "SpinnerButton" component for our library based on the "Interactive Integration" button from MUI here: https://material-ui.com/components/progress/ (but without the circle icon button, just the text version). it gives the same feedback without more custom code.

Reply:
This is pretty much exactly what that example you linked to already does, I think

Reply:
happy to separate this out as another component though